### PR TITLE
feat(medusa): complete cart create reservation

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**

### CODE REVIEW

File: packages/core/core-flows/src/definition/cart/steps/reserve-inventory.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    const inventoryService = container.resolve<IInventoryServiceNext>(
+      ModuleRegistrationName.INVENTORY
+    )
+
+    const items = data.items.map((item) => ({
+      line_item_id: item.id,
+      inventory_item_id: item.inventory_item_id,
+      quantity: item.required_quantity * item.quantity,
+      allow_backorder: item.allow_backorder,
+      location_id: item.location_ids[0],
```
Comment (olivermrbl) [inline packages/core/core-flows/src/definition/cart/steps/reserve-inventory.ts:29]: Why does the step accept multiple location IDs, if we always use only one?

File: packages/core/core-flows/src/order/workflows/create-orders.ts [hunk 2]
```diff
@@ -39,8 +38,8 @@ export const createOrdersWorkflow = createWorkflow(
       findOrCreateCustomerStep({
         customerId: input.customer_id,
         email: input.email,
-      }),
-      validateVariantsExistStep({ variantIds })
+      })
+      // validateVariantsExistStep({ variantIds })
```
Comment (olivermrbl) [inline packages/core/core-flows/src/order/workflows/create-orders.ts:42]: should this be removed?

**Review by olivermrbl — APPROVED** [review]
LGTM

File: packages/core/core-flows/src/order/workflows/create-orders.ts (hunk 2 above)
Comment (carlos-r-l-rodrigues) [reply packages/core/core-flows/src/order/workflows/create-orders.ts:42]: I will refactor the order workflows to match all the changes done in carts. If you don't mind I will keep this commented out for now.

The RemoteQuery will validate if the ids exist or not, but it should be possible to customize the message thrown by the RemoteQuery.

File: packages/core/core-flows/src/definition/cart/steps/reserve-inventory.ts (hunk 1 above)
Comment (carlos-r-l-rodrigues) [reply packages/core/core-flows/src/definition/cart/steps/reserve-inventory.ts:29]: Currently this is the behavior, [to get first location](https://github.com/medusajs/medusa/blob/884428a1b573e499d7659aefed639bf797147428/packages/medusa/src/services/product-variant-inventory.ts#L539).
But I believe this should be reviewed to allocate the stock in all available locations until it matches the quantity.

File: packages/core/core-flows/src/order/workflows/create-orders.ts (hunk 2 above)
Comment (olivermrbl) [reply packages/core/core-flows/src/order/workflows/create-orders.ts:42]: Sure, let's keep it in

File: packages/core/core-flows/src/definition/cart/steps/reserve-inventory.ts (hunk 1 above)
Comment (olivermrbl) [reply packages/core/core-flows/src/definition/cart/steps/reserve-inventory.ts:29]: Yeah I agree. We might even need to look into stock allocation strategies, so merchants can configure exactly how an order is fulfilled and from where

**Pull request merged**